packages/hash/api/src/model/knowledge/hashInstance.model.ts

```diff
@@ -115,18 +118,47 @@ export default class extends EntityModel {
   ): Promise<boolean> {
     const { userModel } = params;
 
-    return await LinkModel.get(graphApi, {
-      sourceEntityId: this.entityId,
-      linkTypeId: SYSTEM_TYPES.linkType.admin.schema.$id,
-      targetEntityId: userModel.entityId,
-    })
-      .then(() => true)
-      .catch((error: Error) => {
-        if (error instanceof NotFoundError) {
-          return false;
-        }
-        throw error;
-      });
+    const outgoingAdminLinkEntityModels = await LinkEntityModel.getByQuery(
```

Comment:
I wonder if we want to make the `LinkEntityModel` `getOutgoingLink` filter something we can return rather than directly use in the method. Then we could use that one, add a few more filters here (e.g. the type constraint) and then query with it.

I hope that made sense, not asking for a change though, just an idea.

Reply:
Hmm that's interesting, easiest thing to slightly improve this here is just to make use of `.getOutgoingLinks` directly like so https://github.com/hashintel/hash/pull/1435/commits/d464460c712666526e01ecf44ef08ed4fa683e85. This also addresses re-usability, but is a slightly different approach.